eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -288,8 +288,28 @@ public function locationDeleted($locationId)
         $this->logger->logCall(__METHOD__, array('location' => $locationId));
         $return = $this->persistenceHandler->urlAliasHandler()->locationDeleted($locationId);
 
-        $this->cache->clear('urlAlias', 'location', $locationId);
+        $this->clearLocation($locationId);
 
         return $return;
     }
+
+    /**
+     * @param $locationId
+     */
+    protected function clearLocation($locationId)
+    {
+        $locationCache = $this->cache->getItem('urlAlias', 'location', $locationId);
+
+        if ($locationCache->isMiss()) {
+            // we need to clear all if we don't have location id in cache
+            $this->cache->clear('urlAlias');
+        } else {
+            $urlAliasesId = $locationCache->get();
+
+            foreach ((array)$urlAliasesId as $eachUrlAliasId) {
```

Comment:
The cast is here, because I don't know if the value is stored always in array so to ensure this foreach method works as expected I casted it, I will remove it.